contracts/impls/centralized/WitnetRequestBoard.sol

```diff
@@ -0,0 +1,278 @@
+import "../../utils/Upgradable.sol";
+
+// Uses:
+import "../../exports/WitnetRequest.sol";
+
+/**
+ * @title Witnet Requests Board mocked
+ * @notice Contract to bridge requests to Witnet for testing purposes.
+ * @dev This contract enables posting requests that Witnet bridges will insert into the Witnet network.
+ * The result of the requests will be posted back to this contract by the bridge nodes too.
+ * @author Witnet Foundation
+ */
+contract WitnetRequestBoard
+    is 
+        Destructible,
+        Upgradable, 
+        WitnetBoard, 
+        WitnetBoardDataWhitelists
+{
+    uint256 internal constant __ESTIMATED_REPORT_RESULT_GAS = 102496;
+
+    constructor() {
+        // sets instance as initialized:
+        __data().instance = address(this);
+    }
+
+
+    // ================================================================================================================
+    // --- Overrides 'Destructible' -----------------------------------------------------------------------------------
+
+    function destroy() external override onlyOwner {
+        selfdestruct(payable(msg.sender));
+    }
+
+
+    // ================================================================================================================
+    // --- Overrides 'Upgradable' -------------------------------------------------------------------------------------
+
+    /// @dev Initialize storage-context when invoked as delegatecall. 
+    /// @dev Should fail when trying to initialize same instance more than once.
+    function initialize(bytes memory _initData) virtual external override {
+        address _owner = __data().owner;
+        if (_owner == address(0)) {
+            // set owner if not yet set
+            _owner = msg.sender;
+            __data().owner = _owner;
+        }
+
+        // only owner can initialize:
+        require(msg.sender == _owner, "WitnetRequestBoard: only owner");
+
+        // current instance cannot be initialized more than once:
+        require(__data().instance != __stub, "WitnetRequestBoard: already initialized");
+        __data().instance = __stub;
+
+        // do actual initialization:
+        setReporters(abi.decode(_initData, (address[])));
+        emit Initialized(msg.sender, __stub);
+    }
+
+    /// @dev Retrieves named version of current implementation.
+    function version() virtual external pure override returns (string memory) {
+        return "WitnetRequestBoard-Centralized-v0.3.0";
+    }
+
+
+    // ================================================================================================================
+    // --- Utility functions not declared within an interface ---------------------------------------------------------
+
+    function setReporters(address[] memory reporters)
+        public
+        virtual
+        onlyOwner
+    {
+        for (uint ix = 0; ix < reporters.length; ix ++) {
+            address _reporter = reporters[ix];
+            if (_reporter != address(0)) {
```

Comment:
Why this check?

Reply:
Indeed, it may not be needed in current implementation, as the reporters list is only used for checking ` msg.sender`, which it's true can never be zero. Removing this check on elements of passed array decreases gas cost of calling `setReporters(address[])` with large arrays.